### ISSUE TITLE
Intermittent Integration Test failures

### ISSUE THREAD

**Issue by PlagueHO** [PowerShell/DscResources#172]
There is an issue where Integration tests intermittently fail (about 1 in 25 test runs) because the DSC LCM is executing another consistency check:
![2016-08-09_20-17-22](https://cloud.githubusercontent.com/assets/7589164/17509556/a47df204-5e6e-11e6-924a-31b33fd0a6c5.png)

See: https://ci.appveyor.com/project/PowerShell/xnetworking/build/2.10.334.0

Triggering AppVeyor to build again resolves the issue. However, for non PowerShell team members this requires a dummy commit (which isn't ideal).

@narrieta noticed this problem in xNetworking (I've seen it many times in various repos that have integration tests and have just gotten used to working around it). I know others have seen it too. Might be time to look into a proper solution.

I think this is most likely because an LCM consistency check has been kicked off by the ConfigurationMode timer. I will try and do some more testing to see what how the LCM is configured on an AppVeyor base OS. I find it unlikely that the LCM ConfigurationMode timer would kick off so soon after a config was applied.

We could probably work around this by adding the -Force parameter to the `{ Get-DscConfiguration -Verbose -ErrorAction Stop } | Should Not throw` but this bothers me a bit - feels like a work around rather than a solution.

Any thoughts from the community and the MSFT DSC team would be appreciated.

**Comment by BrianFarnhill** [comment]
Well we do have admin rights on the appveyor boxes so we should be able to configure the LCMs how we need them right?

I've still been trying to get my integration tests sorted for SharePointDsc but what we were looking at (and it's yet to be tested yet) was the concept of using the AfterEach{} block in Pester to clear the configuration documents with Remove-DscConfigurationDocument to make sure that if that job or anything else was to to fire while we weren't looking it wouldn't have any docs to apply. Also menat we could have a clean slate at the start of each run. Worth looking in to maybe? Thoughts?

**Comment by PlagueHO** [comment]
@BrianFarnhill - some good ideas there I reckon.

It might actually be worth adding a function to the DSCResource.Tests TestHelpers.psm1 to reset the LCM and clear out the documents.

e.g.
```powershell
function Reset-DSC {
    Stop-DscConfiguration -Force -ErrorAction SilentlyContinue
    Remove-DscConfigurationDocument -Stage Current -Force
}
```
This could then be called automatically by Initialize-TestEnvironment and Restore-TestEnvironment (for integration test types only) - as well as in AfterEach{} blocks.

What do you reckon?

**Comment by kilasuit** [comment]
@PlagueHO that sounds like a good idea to me :-) :+1:

**Comment by BrianFarnhill** [comment]
@PlagueHO Yea I agree, I would remove more than just the Current document, take the previous and pending ones too incase things go south in the tests - but that is a much better idea.

We definitely want it as a separate function though and don't just rely on it in the initialise though, because if I have multiple configs to apply in one describe block (like a first one to create and object and a second one to update it or something) then we still wanna be able to clear it between each of those tests, but outside of that scenario it's a top idea.

**Comment by PlagueHO** [comment]
@BrianFarnhill, @kilasuit  - cool. I'll submit a PR to the DSCResource.tests module and we can go from there. I'll tag this issue in the PR so you can comment.

Good idea about the deleting the pending and previous documents as well.

**Comment by PlagueHO** [comment]
I'll close this one now. If anyone runs into a problem with an intermittent IT failure we can reopen it.

**Comment by PlagueHO** [comment]
Unfortunately we just ran into this issue again in xNetworking (see https://ci.appveyor.com/project/PowerShell/xnetworking/build/2.12.367.0):
![2016-10-24_14-03-24](https://cloud.githubusercontent.com/assets/7589164/19631069/527f4334-99f3-11e6-9514-e4a9428ae170.png)

It looks like [this PR](https://github.com/PowerShell/DscResource.Tests/pull/72) didn't resolve the problem. So, I've reopened this issue.

One other solution may be to call ```Get-DSCConfigurationStatus``` and before [this line](https://github.com/PowerShell/xNetworking/blob/dev/Tests/Integration/MSFT_xDNSServerAddress.Integration.Tests.ps1#L40) is called and wait for ```Status -ne 'Running'```.

If this was done I'd suggest adding a function to the TestHelpers.psm1 module (e.g. ```Wait-DSC```) that waits for the LCM to complete.

The problems with the above approach are:
1. If the LCM is stalled then the tests will never complete - perhaps a timeout would solve this?
2. Each integration test would need to be modified to include this call to ```Wait-DSC```.

I know this is a fairly minor issue but it is a pain to resolve when it occurs (at least for us non-MSFTies :grin:).
